src/services/checkS3BucketData.spec.ts

```diff
@@ -120,20 +161,129 @@ describe('check objects in analysis bucket', () => {
         'firehose/2022/10/10/23/example-object-3'
       ]
     })
+    assertNumberOfFilesLogged(9, 0)
   })
 
-  test('partial data in analysis bucket', async () => {
-    givenPartialDataInAnalysisBucket(prefixes)
+  test('no data in analysis bucket, audit bucket data contains some data with missing keys', async () => {
+    givenDataInBucketForPrefixes(
+      [prefixes[0], prefixes[2]],
+      TEST_AUDIT_BUCKET,
+      'STANDARD'
+    )
+    givenDataInBucketForPrefix(prefixes[1], TEST_AUDIT_BUCKET, [
+      { StorageClass: 'STANDARD' }
```

Comment:
Is it also worth adding an object with `{ StorageClass: 'STANDARD', Key: undefined }`

Reply:
Done